### DISCUSSION TITLE
LDM Notes for October 2nd, 2023

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7573] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-10-02.md

## Agenda

- Collection expressions

**Comment by Serdan** [comment]
Anyone have a link to that horrifying picture of the IEnumerable hierarchy? 😄

**Reply by jnm2** [reply]
Yes :D

<img src="https://github.com/dotnet/csharplang/assets/8040367/7c2672e6-69f3-4147-a8d4-b2d28d955fb9" width="50%">

I didn't share this one during LDM, but the dictionary version:

![image](https://github.com/dotnet/csharplang/assets/8040367/0c29b88c-bf6a-4f43-b32a-9189ce368d5f)

(I'd have posted the mermaid source, but I haven't figured out how to include angle brackets in mermaid-inside-markdown-code-block without github render errors.)

**Reply by AartBluestoke** [reply]
for "clarification" i've put the "commonly used collections" (https://learn.microsoft.com/en-us/dotnet/standard/collections/commonly-used-collection-types)  onto that chart:
![image](https://github.com/dotnet/csharplang/assets/1892829/e22df364-f247-46e9-b216-194f47dda8a2)
[mermaid Chart link](https://mermaid.live/edit#pako:eNqNU8GKwjAU_JWQg-yC_kCFhSXroRBQ2x57iclTi2kiacIi4r9vk7bW1uJKD2ky82Ze5rVXzLUAHGG0l_qXH5mxCNFkmSuEUOV2B8PORxQrC2bPOFTh_AFRWi0OoMAUvIUQiomWErgttEKLxReKV8qVYNhOQs-hRWUbtGe3KCgxthlaxCT7yPFD4UzaZTY72GWOP1vLrOOuArdvYYI7bi-moca3-Mwmd-VkTTwtASbWSl7eaydZ08eiaY9aueNvSHPZjdHCcTBEq8q3-56br_435bB023vkRLKqug8coW9j2KXR8k0Pz_tp3rGtAweT3qll_DSJhNyfImmhVj8bbOs4h3BnT1stdQLxcpCUTidEewsySkYr7owB1TiRrXfqD8PFn-38KDr-UJikQ4GQzyuBdCTQzS6s4c0v_sFzXH8rJStE_YtfPZRje4QSchzVr4KZU45zdat5zFmdXhTHkTUO5tidBbPwU7D60iWO9kxWcPsDszlCSw)
(a few things like 'List<T>' directly implement IList<T> and ICollection<T> - i've only linked to the derived type

**Reply by DavidArno** [reply]
@AartBluestoke, you now just need to show that `Array` pseudo-implements those top four generic interfaces at runtime (but only when single dimensional)  and we are then getting close to the true horror of this hierarchy 😄
